history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskCancelClaimAccTest.java

```diff
@@ -28,66 +32,85 @@ class CreateHistoryEventOnTaskCancelClaimAccTest extends AbstractAccTest {
-
-    final String taskId = "TKI:000000000000000000000000000000000043";
-    Task task = taskService.getTask(taskId);
-    final Instant oldModified = task.getModified();
-    final Instant oldClaimed = task.getClaimed();
-
-    TaskHistoryQueryMapper taskHistoryQueryMapper = getHistoryQueryMapper();
-    List<TaskHistoryEvent> events =
-        taskHistoryQueryMapper.queryHistoryEvents(
-            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(taskId));
-
-    assertThat(events).isEmpty();
-
-    assertThat(task.getState()).isEqualTo(TaskState.CLAIMED);
-    task = taskService.forceCancelClaim(taskId);
-    assertThat(task.getState()).isEqualTo(TaskState.READY);
-
-    events =
-        taskHistoryQueryMapper.queryHistoryEvents(
-            (TaskHistoryQueryImpl) historyService.createTaskHistoryQuery().taskIdIn(taskId));
-
-    assertThat(events).hasSize(1);
-
-    TaskHistoryEvent event = events.get(0);
-
-    assertThat(event.getEventType()).isEqualTo(TaskHistoryEventType.CLAIM_CANCELLED.getName());
-
-    event = historyService.getTaskHistoryEvent(event.getId());
-
-    assertThat(event.getDetails()).isNotNull();
-
-    JSONArray changes = new JSONObject(event.getDetails()).getJSONArray("changes");
-
-    JSONObject expectedClaimed =
-        new JSONObject()
-            .put("newValue", "")
-            .put("fieldName", "claimed")
-            .put("oldValue", oldClaimed.toString());
-    JSONObject expectedModified =
-        new JSONObject()
-            .put("newValue", task.getModified().toString())
-            .put("fieldName", "modified")
-            .put("oldValue", oldModified.toString());
-    JSONObject expectedState =
-        new JSONObject()
-            .put("newValue", "READY")
-            .put("fieldName", "state")
-            .put("oldValue", "CLAIMED");
-    JSONObject expectedOwner =
-        new JSONObject().put("newValue", "").put("fieldName", "owner").put("oldValue", "user-b-1");
-
-    JSONArray expectedChanges =
-        new JSONArray()
-            .put(expectedClaimed)
-            .put(expectedModified)
-            .put(expectedState)
-            .put(expectedOwner);
-
-    assertThat(changes.similar(expectedChanges)).isTrue();
+  @TestFactory
+  Stream<DynamicTest> should_CreateCancelClaimedHistoryEvent_When_TaskIsCancelClaimed() {
+
+    List<Triplet<String, String, String>> list =
+        List.of(
+            Triplet.of(
+                "With Attachment and secondary Object Reference",
+                "TKI:000000000000000000000000000000000002",
+                "user-1-1"),
+            Triplet.of(
+                "Without Attachment and secondary Object References",
+                "TKI:000000000000000000000000000000000043",
+                "user-b-1"));
+    ThrowingConsumer<Triplet<String, String, String>> test =
+        t -> {
+          Task task = taskService.getTask(t.getMiddle());
```

Comment:
suggestion: give t.getMiddle() and t.getRight() a name by assigning them to a variable.
This might make the test more readable since the reader doesn't need to ask himself what middle and right are.

Reply:
Ok :)